rust/rsmgp-sys/src/result/mod.rs

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2016-2021 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use snafu::Snafu;
+use std::ffi::CStr;
+
+use crate::edge::*;
+use crate::list::*;
+use crate::map::*;
+use crate::memgraph::*;
+use crate::mgp::*;
+use crate::path::*;
+use crate::value::*;
+use crate::vertex::*;
+// Required here, if not present tests linking fails.
+#[double]
+use crate::mgp::ffi;
+use mockall_double::double;
+
+pub struct MgpResultRecord {
+    ptr: *mut mgp_result_record,
+    memgraph: Memgraph,
+}
+
+impl MgpResultRecord {
+    pub fn new(memgraph: &Memgraph) -> MgpResult<MgpResultRecord> {
```

Comment:
I think this function is kind different from e.g.: `Vertex::new` because this function creates a new object, while `Vertex::new` only wraps the pointer. I would use different names for them, maybe using `create` and `wrap` explicitly, but that is a details. The important in my opinion is to differentiate between the two types of functions.